05-FITS/io_fits_ascii_overview.ipynb

```diff
@@ -480,94 +475,98 @@
+    "fig, ax = plt.subplots()\n",
+    "ax.hist(tab['energy'].value, bins=128, log=True)\n",
+    "ax.semilogy()\n",
+    "ax.set_xlabel(\"eV\")\n",
+    "\n",
+    "# The trailing \";\" suppresses extraneous matplotlib output in notebook.\n",
+    "ax.set_ylabel(\"Counts per bin\");"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "<div class=\"alert alert-block alert-info\">\n",
-    "<h3>Exercise:</h3>\n",
-    "<br>\n",
-    "Construct programmatically a FITS file with 1 image extension \n",
-    "and save it to disk. Use a numpy array with random numbers as \n",
-    "data.\n",
-    "\n",
-    "Hints: \n",
-    "- Use `fits.HDUList()` to create a FITS file.\n",
-    "- Use `fits.PrimaryHDU()` for the primary HDU and `fits.ImageHDU()` for the image HDU. \n",
-    "\n",
-    "To look at the documentation of a python object, type one of the following into a python or IPython terminal:\n",
-    "\n",
-    "    >>> help(fits.ImageHDU)\n",
-    "    >>> fits.ImageHDU?\n",
-    "\n",
-    "</div>"
+    "#### 2D Histogram"
    ]
   },
   {
-   "cell_type": "markdown",
+   "cell_type": "code",
+   "execution_count": null,
    "metadata": {},
+   "outputs": [],
    "source": [
-    "---\n",
-    "# Working with ASCII files"
+    "from matplotlib.colors import LogNorm\n",
+    "\n",
+    "fig, ax = plt.subplots()\n",
+    "h = ax.hist2d(tab['x'].value, tab['y'].value,\n",
+    "              bins=(1000, 1000), norm=LogNorm())\n",
+    "fig.colorbar(h[3]);"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "What if your data is in a text file? We've included `throughput.txt` as an example."
+    "### Working with large files"
    ]
   },
   {
-   "cell_type": "code",
-   "execution_count": null,
+   "cell_type": "markdown",
    "metadata": {},
-   "outputs": [],
    "source": [
-    "! cat throughput.txt"
+    "The `open()` function supports a `memmap=True` option that allows the array data of each HDU to be accessed with memory mapping, rather than being read into memory all at once. This is particularly useful for working with very large arrays that cannot fit entirely into physical memory."
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "Python has a built-in method for opening text files like ASCII files, but it's not too friendly. Let's see how it works:"
+    "---\n",
+    "# Working with ASCII files"
    ]
   },
   {
-   "cell_type": "code",
-   "execution_count": null,
+   "cell_type": "markdown",
    "metadata": {},
-   "outputs": [],
    "source": [
-    "with open('throughput.txt') as f:\n",
-    "    throughput_data = f.read()\n",
-    "throughput_data"
+    "What if your data is in a text file? We've included `throughput.txt` as an example.\n",
+    "\n",
+    "    Detector throughput values\n",
+    "    â€”â€”\n",
```

Comment:
```suggestion
    "    --\n",
```

Reply:
I copy-paste this from the data file. Are you seeing something different when you open `throughput.txt`? I wonder if the encoding is OS-specific.

Reply:
OK, nvm... Now I see the dashes. :woman_shrugging: